boa/src/builtins/object/mod.rs

```diff
@@ -67,6 +68,7 @@ pub struct Object {
 #[derive(Debug, Trace, Finalize, Clone)]
 pub enum ObjectData {
     Array,
+    Map(GcCell<OrderedMap<Value, Value>>),
```

Comment:
I don't think we need to `GcCell` since we expanded `get_map` in to it's methods, this would eliminate the checks from `GcCell`.